test/suites/test_push.py

```diff
@@ -0,0 +1,449 @@
+        on_replace_callback
+    )
+    """)
+
+    return srv
+
+
+# Callback for on_push arg (for testing purposes).
+def push_callback(data, on_push_ctx=[]):
+                data[0][1] = data[0][1] + 1
+                on_push_ctx.append(data)
+
+
+@unittest.skipIf(sys.platform.startswith("win"),
+                 'Mesh tests on windows platform are not supported')
+class TestSuite_Push(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(self):
+        print(' PUSH '.center(70, '='), file=sys.stderr)
+        print('-' * 70, file=sys.stderr)
+
+    def setUp(self):
+        # Create server and extract helpful fields for tests.
+        self.srv = create_server()
+        self.host = self.srv.host
+        self.port = self.srv.args['primary']
+
+    def test_00_00_push_call_via_connect(self):
+        def assert_push_call(conn):
+            callback_res = []
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=NetworkWarning)
+                resp = conn.call(
+                    'server_function',
+                    on_push=push_callback, 
+                    on_push_ctx=callback_res
+                )
+            self.assertEqual(callback_res, [[[1, 1]], [[2, 1]], [[3, 1]]])
+            self.assertEqual(resp.data[0], [3, 0])
+
+        conn = tarantool.connect(host=self.host, port=self.port, user='test', password='test')
+        # Check push from instance.
+        assert_push_call(conn)
+        conn.close()
+        # Stop instance.
+        self.srv.stop()
+
+        with self.assertRaises(NetworkError):
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=NetworkWarning)
+                conn.ping()
+
+    def test_00_01_push_eval_via_connect(self):
+        def assert_push_eval(conn):
+            callback_res = []
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=NetworkWarning)
+                resp = conn.eval(
+                    'return server_function()',
+                    on_push=push_callback, 
+                    on_push_ctx=callback_res
+                )
+            self.assertEqual(callback_res, [[[1, 1]], [[2, 1]], [[3, 1]]])
+            self.assertEqual(resp.data[0], [3, 0])
+
+        conn = tarantool.connect(host=self.host, port=self.port, user='test', password='test')
+        # Check push from instance.
+        assert_push_eval(conn)
+        conn.close()
+        # Stop instance.
+        self.srv.stop()
+
+        with self.assertRaises(NetworkError):
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=NetworkWarning)
+                conn.ping()
+
+    
+    def test_00_02_push_insert_via_connect(self):
+        def assert_push_insert(conn):
+            callback_res = []
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=NetworkWarning)
```

Comment:
And why we should receive any `NetworkWarning`s here?

Reply:
Done, deleted